mirascope/mcp/client.py

```diff
@@ -0,0 +1,280 @@
+import base64
+import string
+from collections.abc import Awaitable, Callable
+from typing import Any, ClassVar, TypeVar
+
+from mcp import ClientSession
+from mcp.types import (
+    BlobResourceContents,
+    CallToolResult,
+    EmbeddedResource,
+    ImageContent,
+    Prompt,
+    PromptMessage,
+    Resource,
+    TextContent,
+    TextResourceContents,
+    Tool,
+)
+from pydantic import AnyUrl, BaseModel, ConfigDict, Field, create_model
+
+from mirascope.core import BaseMessageParam, BaseTool
+from mirascope.core.base import AudioPart, DocumentPart, ImagePart
+from mirascope.core.base._utils import convert_base_model_to_base_tool
+
+_BaseToolT = TypeVar("_BaseToolT", bound=BaseTool)
+
+
+class MCPClientTool(BaseTool):
```

Comment:
I noticed that Client and Server handle Tool differently, so I need a dedicated tool for each.